### PR TITLE
Handle empty slices correctly

### PR DESCRIPTION
**Pull request by sdroege**
Passing `NULL` to `slice::from_raw_parts` is invalid.

### CODE REVIEW

**Comment by sdroege** [comment]
@bilelmoussaoui Something broke with the GTK version in the images

**Comment by bilelmoussaoui** [comment]
> @bilelmoussaoui Something broke with the GTK version in the images

It is caused by GDK_EVENT_LAST, see #864

**Comment by bilelmoussaoui** [comment]
If you rebase, the CI should be happy

File: gtk4/src/editable.rs [hunk 1]
```diff
@@ -45,7 +45,9 @@ unsafe extern "C" fn insert_text_trampoline<T, F: Fn(&T, &str, &mut i32) + 'stat
 ) where
     T: IsA<Editable>,
 {
-    let buf = if new_text_length != -1 {
+    let buf = if next_text_length == 0 {
```
Comment (bilelmoussaoui) [inline gtk4/src/editable.rs:48]:
```suggestion
    let buf = if new_text_length == 0 {
```

**Pull request merged**